core/src/main/java/org/apache/iceberg/PartitionSpecParser.java

```diff
@@ -98,7 +101,16 @@ static void toJsonFields(UnboundPartitionSpec spec, JsonGenerator generator) thr
       generator.writeStartObject();
       generator.writeStringField(NAME, field.name());
       generator.writeStringField(TRANSFORM, field.transformAsString());
-      generator.writeNumberField(SOURCE_ID, field.sourceId());
+      if (field.sourceIds().size() == 1) {
+        generator.writeNumberField(SOURCE_ID, field.sourceId());
+      } else {
+        generator.writeFieldName(SOURCE_IDS);
+        generator.writeStartArray();
+        for (Integer value : field.sourceIds()) {
+          generator.writeNumber(value);
+        }
+        generator.writeEndArray();
+      }
```

Comment:
nit: Iceberg follows the new line spacing for code blocks 
https://iceberg.apache.org/contribute/?h=contribut#block-spacing

Reply:
Ack, fixing.